sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/ResolvedDataSource.scala

```diff
@@ -92,19 +96,61 @@ object ResolvedDataSource extends Logging {
     }
   }
 
+  // TODO: Combine with apply?
   def createSource(
       sqlContext: SQLContext,
       userSpecifiedSchema: Option[StructType],
       providerName: String,
       options: Map[String, String]): Source = {
     val provider = lookupDataSource(providerName).newInstance() match {
-      case s: StreamSourceProvider => s
+      case s: StreamSourceProvider =>
+        s.createSource(sqlContext, userSpecifiedSchema, providerName, options)
+
+      case format: FileFormat =>
+        val caseInsensitiveOptions = new CaseInsensitiveMap(options)
+        val path = caseInsensitiveOptions.getOrElse("path", {
+          throw new IllegalArgumentException("'path' is not specified")
+        })
+        val metadataPath = caseInsensitiveOptions.getOrElse("metadataPath", s"$path/_metadata")
+
+        val allPaths = caseInsensitiveOptions.get("path")
+        val globbedPaths = allPaths.toSeq.flatMap { path =>
```

Comment:
Good catch.  I think the next step here is to actually try and combine the logic for streaming and batch file sources as the TODO above says.